egg/core/callbacks.py

```diff
@@ -169,6 +182,30 @@ def get_checkpoint(self):
             optimizer_state_dict=self.trainer.optimizer.state_dict(),
         )
 
+    def get_checkpoint_files(self):
+        """
+        Return a list of the files in the checkpoint dir
+        """
+        return [name for name in os.listdir(self.checkpoint_path) if ".tar" in name]
+
+    @staticmethod
+    def natural_sort(to_sort):
+        """
+        Sort a list of files naturally
+        E.g. [file1,file4,file32,file2] -> [file1,file2,file4,file32]
+        """
+        convert = lambda text: int(text) if text.isdigit() else text.lower()
+        alphanum_key = lambda key: [convert(c) for c in re.split("([0-9]+)", key)]
+        return sorted(to_sort, key=alphanum_key)
```

Comment:
if you resume a training there might be a checkpoint ending with _final_, see [here](https://github.com/facebookresearch/EGG/blob/6e029ca514d5f0df102ff45012698caf0b6dedf9/egg/core/callbacks.py#L163-L166)

Reply:
the final version is the first of the sorted list